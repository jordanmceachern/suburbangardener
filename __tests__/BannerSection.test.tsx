import { render } from "@testing-library/react";
import BannerSection from "@/components/BannerSection";

describe("BannerSection Component", () => {
  it("should match DOM snapshot", () => {
    const { container } = render(<BannerSection />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should maintain CSS classes for banner layout", () => {
    const { container } = render(<BannerSection className="custom-banner" />);

    // Check for banner-related classes
    const hasBannerClasses =
      container.innerHTML.includes("flex") &&
      container.innerHTML.includes("items-center") &&
      container.innerHTML.includes("text-center");

    expect(hasBannerClasses).toBe(true);
    expect(container.innerHTML).toContain("custom-banner");
  });
});
