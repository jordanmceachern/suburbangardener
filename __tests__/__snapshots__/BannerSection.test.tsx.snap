// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`BannerSection Component should match DOM snapshot 1`] = `
<header
  class="relative flex flex-col items-center text-center overflow-hidden rounded-lg min-h-[400px] "
>
  <video
    autoplay=""
    class="absolute inset-0 w-full h-full object-cover z-0"
    loop=""
    playsinline=""
    poster="https://res.cloudinary.com/dtweazqf2/image/upload/q_auto,f_auto/v1754661544/SuburbanGardener/IMG_7791_nl7dzw.jpg"
  >
    <source
      src="https://res.cloudinary.com/dtweazqf2/video/upload/f_auto,q_auto/v1754668830/SuburbanGardener/bannerboomerang_ebv27k.webm"
      type="video/webm"
    />
    Your browser does not support the video tag.
  </video>
  <div
    class="absolute inset-0 z-10 w-full flex flex-col items-center justify-center"
  >
    <div
      class="h-[125px] sm:h-[225px] w-full flex"
    >
      <div
        class="h-auto w-full flex justify-center"
      >
        <div
          aria-label="Suburban Gardener Logo"
          class="bg-contain bg-center bg-no-repeat w-full h-full  rounded-lg"
          style="background-image: url("https://res.cloudinary.com/dtweazqf2/image/upload/c_fill,t_transparent-white-alpha-dim,q_auto,f_auto/v1755011510/SuburbanGardener/suburban_gardener_s4c8gy.png");"
        />
      </div>
    </div>
  </div>
</header>
`;
